Microsoft.Research/Clousot.Cache/SQLCacheModel.cs

```diff
@@ -784,5 +786,33 @@ protected override SqlConnectionStringBuilder BuildConnectionString(IClousotCach
 
       return base.BuildConnectionString(options);
     }
+
+    public static string GetDesiredDbName()
+    {
+        var versionKeyFileMapping = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
```

Comment:
Shoulw we use static field instead? Or even path this as an argument to the GetDesiredDbName?

When I've made this change I tried to separate responsibilities and extracted all "application specific" functionality into high level. That's why I've added dbname as parameter for the constructor.

BTW there is at least one another client of this parameterless constructor and I don't know is this functionality required there or not. Thats why I would suggest to put "desired db name selection" out of this class.